homeassistant/components/energy/data.py

```diff
@@ -54,18 +54,38 @@ class FlowToGridSourceType(TypedDict):
     stat_compensation: str | None
 
     # Used to generate costs if stat_compensation is set to None
-    entity_energy_from: str | None  # entity_id of an energy meter (kWh), entity_id of the energy meter for stat_energy_from
+    entity_energy_to: str | None  # entity_id of an energy meter (kWh), entity_id of the energy meter for stat_energy_to
     entity_energy_price: str | None  # entity_id of an entity providing price ($/kWh)
     number_energy_price: float | None  # Price for energy ($/kWh)
 
 
+class FlowNetGridSourceType(TypedDict):
+    """Dictionary describing a 'net' stat for the grid source."""
+
+    # kWh meter
+    stat_energy_net: str
+
+    # statistic_id of net compensation ($)
+    # If set to None and entity_energy_from and entity_energy_price are configured,
+    # an EnergyCostSensor will be automatically created
+    stat_cost: str | None
+
+    # Used to generate costs if stat_compensation is set to None
+    entity_energy_net: str | None  # entity_id of an energy meter (kWh), entity_id of the energy meter for stat_energy_net
+    entity_energy_price_to: str | None
+    number_energy_price_to: str | None
+    entity_energy_price_from: str | None
+    number_energy_price_from: str | None
+
+
 class GridSourceType(TypedDict):
     """Dictionary holding the source of grid energy consumption."""
 
     type: Literal["grid"]
 
     flow_from: list[FlowFromGridSourceType]
     flow_to: list[FlowToGridSourceType]
+    flow_net: list[FlowNetGridSourceType]
```

Comment:
Interesting. I was thinking we would just store "stat X, upwards" in the original stat field. Now we can't use net values for Solar or battery without updating their formats too

Reply:
we can use net meters for solar or battery, but we can't use *only increase* or *only decrease* for solar or battery